src/app/app.component.html

```diff
@@ -17,178 +31,193 @@
+              <span class="pficon pficon-replicator" data-toggle="tooltip" title="Templates"></span>
+              <span class="list-group-item-value">Templates</span>
+            </a>
+          </li>
+          -->
+          <li class="list-group-item"
+              routerLinkActive="active">
+            <a [routerLink]=" ['connections'] ">
+            <span class="pficon pficon-network"
+                  data-toggle="tooltip"
+                  title="Connections"></span>
+              <span class="list-group-item-value">Connections</span>
+            </a>
+          </li>
+          <li class="list-group-item"
+              routerLinkActive="active">
+            <a [routerLink]=" ['settings'] ">
+            <span class="fa fa-lock"
+                  data-toggle="tooltip"
+                  title="Settings"></span>
+              <span class="list-group-item-value">Settings</span>
+            </a>
           </li>
-        -->
         </ul>
-      </nav>
-    </nav>
-    <!--/.navbar-->
-  </header>
-
-  <!-- Sidebar -->
-  <div class="nav-pf-vertical hidden-icons-pf"
-       *ngIf="loggedIn">
-    <ul class="list-group">
-      <li class="list-group-item"
-          routerLinkActive="active">
-        <a [routerLink]=" ['dashboard'] ">
-          <span class="fa fa-tachometer"
-                data-toggle="tooltip"
-                title="Dashboard"></span>
-          <span class="list-group-item-value">Home</span>
-        </a>
-      </li>
-      <li class="list-group-item"
-          routerLinkActive="active">
-        <a [routerLink]=" ['integrations'] ">
-          <span class="pficon pficon-topology"
-                data-toggle="tooltip"
-                title="Integrations"></span>
-          <span class="list-group-item-value">Integrations</span>
-        </a>
-      </li>
-      <!--
-      <li class="list-group-item" routerLinkActive="active">
-        <a [routerLink]=" ['templates'] ">
-          <span class="pficon pficon-replicator" data-toggle="tooltip" title="Templates"></span>
-          <span class="list-group-item-value">Templates</span>
-        </a>
-      </li>
-      -->
-      <li class="list-group-item"
-          routerLinkActive="active">
-        <a [routerLink]=" ['connections'] ">
-          <span class="pficon pficon-network"
-                data-toggle="tooltip"
-                title="Connections"></span>
-          <span class="list-group-item-value">Connections</span>
-        </a>
-      </li>
-      <li class="list-group-item"
-          routerLinkActive="active">
-        <a [routerLink]=" ['settings'] ">
-          <span class="fa fa-lock"
-                data-toggle="tooltip"
-                title="Settings"></span>
-          <span class="list-group-item-value">Settings</span>
-        </a>
-      </li>
-    </ul>
+      </div>
+      <main class="container-fluid"
+            [ngClass]="{'container-pf-nav-pf-vertical': !firstTime}"
+            *ngIf="loggedIn">
```

Comment:
Same here, is this needed if we're checking the token in a parent element?  I feel like probably not :-)  Might be worth checking if loggedIn could be refactored out altogether.

Reply:
So, now we are checking for the setup status from the API in addition to the token. I think this app.component.html really needs to be refactored as you say, but perhaps not for TP1.

@akieling Currently the parent element does not take into account the setup status, only whether or not the user has a token, so we are doing that check for individual elements, I guess. There will have to be some redundancy for now.